File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> idx;
        for(int i=0;i<nums.size();i++){
            if(idx.find(target - nums[i])!= idx.end()){
                return {i,idx[target-nums[i]]};
            }
            idx[nums[i]] = i;
        }
        return {0,0};
    }
};

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    idx := make(map[int]int)
    for i:=0 ; i<len(nums) ; i++{
        if id,ok:=idx[target-nums[i]];ok{
            return []int{i,id}
        }
        idx[nums[i]] = i
    }
    return []int{}
}